pkg/prometheus/promcfg.go

```diff
@@ -53,7 +53,7 @@ func sanitizeLabelName(name string) string {
 // compatible with a given Prometheus version.
 type ConfigGenerator struct {
 	logger                 log.Logger
-	version                semver.Version
+	Version                semver.Version
```

Comment:
I'd rather keep it unexported and update the code that rely on it. In practice,

https://github.com/prometheus-operator/prometheus-operator/blob/ef5fad77131af9f162b4b13234446e721d74818d/pkg/prometheus/statefulset.go#L352-L354

This check should be done earlier and/or differently:

```golang
	// Anything before v2.0.0-alpha.0 isn't supported by the operator.
	if cg := cg.WithMinimumVersion("v2.0.0-alpha.0"); !cg.IsCompatible() {
		return nil, errors.Errorf("unsupported Prometheus version %s", cg.Version())
	}

// pkg/prometheus/promcfg.go
func (cg *ConfigGenerator) Version() string {
	return cg.version.String()
}
```

Reply:
Agree we should try to keep the `version` field unexported

Reply:
Looks good!

Small nit that it should return semver.Version because we need to compare `Major` :P